web/src/flow/stages/authenticator_validate/AuthenticatorValidateStage.ts

```diff
@@ -112,6 +115,42 @@ export class AuthenticatorValidateStage
         `);
     }
 
+    willUpdate(_changed: PropertyValues<this>) {
```

Comment:
This code block chooses which `selectedDeviceChallenge` to pick, if any.  I've moved it here because `willUpdate` is a special method; changes to states and properties within it will not trigger a re-draw, so this can be set without causing Lit to draw this component twice.  I also put in early returns to show that it's a selective, progressive algorithm; if one succeeds, a later one doesn't need to be examined.